sdk/storage/azqueue/service/client.go

```diff
@@ -0,0 +1,148 @@
+package service
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azqueue/internal/base"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azqueue/internal/exported"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azqueue/internal/generated"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azqueue/internal/shared"
+	"net/http"
+)
+
+// ClientOptions contains the optional parameters when creating a Client.
+type ClientOptions struct {
+	azcore.ClientOptions
+}
+
+// Client represents a URL to the Azure Queue Storage service allowing you to manipulate queues.
+type Client base.Client[generated.ServiceClient]
```

Comment:
Sorry if this has already been discussed, but can we not directly expose the generated code for queues?  My understanding is that the public surface area is quite small/simple so the generated code should just work.

CC @JeffreyRichter

Reply:
this is pattern matching blobs

Reply:
Blobs is a special case due to the size and complexity of the public surface area.

Reply:
Joel, I think you mean you WANT to expose the code gen directly, right? (your first comment says "can we not directly expose..."). I do think merging the MessageClient into QueueClient is a good thing to do and I don't think the code gen will accomplish this, right? If it doesn' then wrapping may be necessary.

Reply:
Correct, I was hoping we can expose the generated code and not wrap (which is always our default position).  If we can't for reasons, it would be great to enumerate them, so everybody is on the same page.

Reply:
Yes, I agree with you. In this case, the merging of clients may be the reason for wrapping the code-gen.

Reply:
Is it as simple as the APIs being on the same client?  If so, we can fix this via an autorest transform.  I suspect there's more to it than this though.

Reply:
I dont' know all the things that transforms are capable of.  The URL for messages (https://myaccount.queue.core.windows.net/myqueue/messages)  is different from the URL for the queue (https://myaccount.queue.core.windows.net/myqueue) and so the code gen would have to append the "/messages" part to the queue URL. And the method names in swagger are not great (like PUT message instead of Enqueu, for example) but I think this can be fixed with a transform.

Reply:
We have a `QueueClient` and `QueueServiceClient` for the queues package in .NET, Java, Python, and JavaScript.  Why would we diverge from this pattern in Go?  I think it would be more confusing for the customer if we merge them.

Reply:
No, we code gen a MessageClient and this is what we want to merge into QueueClient so there are 2 clients; not 3.

Reply:
Makes sense, we don't have a MessageClient on the other platforms either.

Reply:
I don't think it's necessary to have transforms. Similar to the blob package, we can create a composite client and store an inner client of `messages` alongside `queues`. but unlike blobs there will not be user facing APIs for messages. Instead, the composite client will just directly call the messages generated code as well as the queues generated code. This will all be in a separate PR of course. 

TLDR; we will use the wrapper, which will wrap two clients service client and queue client (composite of queue, message_ID, messages)

In the meantime, if we're good with the service client please approve so I can build on top of it :) @jhendrixMSFT @JeffreyRichter

Reply:
@tasherif-msft does this mean that the merging of the clients means this one will change in a future PR and the `service` package goes away?

Reply:
it will not change. It just means there will be a new package called queue (in addition to service) that will also be wrapped and will include messages and queue ops.

Reply:
How large is the public surface area for each?  Are separate packages really required?  It adds an amount of complexity, both on us and the consumer, so we should be sure that the tradeoff is worth it.

Reply:
Yes, they are required to be separate, this has always been the standard on all languages, Python, Java and .NET the packages mentioned above are exactly the same pattern the other languages are following. Design wise its also preferable to separate account level operations from the queue level operations hence why it's done in all languages and all storage services.

Reply:
offline discussion with @jhendrixMSFT: will go with keeping everything in queue package and having separate clients